### PR TITLE
feat(core): added pathScheme for project generators

### PR DESCRIPTION
**Pull request by rcompton78**
Added pathScheme (default nx) to  app,lib,test generation.  Added and
e2e test and updated generated docs

This addresses issue #427

### CODE REVIEW

File: packages/core/src/generators/app/schema.json
Comment (AgentEnder) [inline packages/core/src/generators/app/schema.json:97]: Lets remove this alias, `-p` frequently refers to a project so this could be confusing.

File: packages/core/src/generators/lib/schema.json
Comment (AgentEnder) [inline packages/core/src/generators/lib/schema.json:79]: Same changes as other schema file.

File: packages/core/src/generators/utils/generate-project.ts
Comment (AgentEnder) [inline packages/core/src/generators/utils/generate-project.ts:67]: This should probably be something like:
```suggestion
  const projectName =
    options.pathScheme === 'nx'
      ? projectDirectory.replace(/\//g, '-')
      : `${names(options.directory).className}.${className}`;
```

I don't really like the way the code looks / readability at this point, but that's more of a comment on the overall state of this normalization method... If you can clean up this little section some it would be appreciated.

File: packages/core/src/generators/utils/generate-test-project.ts
Comment (AgentEnder) [inline packages/core/src/generators/utils/generate-test-project.ts:29]: This will be kinda ugly too, but we should default to `Test` rather than `test` for the suffix when using dotnet path scheme. 

Perhaps:
```typescript
  const suffix = schema.testProjectNameSuffix || (pathScheme === 'nx' ? 'test' : 'Test');
```

I don't really like having the `||` and a ternary here though, that's a lot of complexity represented by the single line. A decent amount of stuff changes based on that too, so it may be easier if we changed to something like:
```typescript
let suffix: string;
let separator: string;
if (schema.pathScheme === 'nx') {
  separator = '-'
  suffix = schema.testProjectNameSuffix || 'test'
} else {
  separator = '.'
  suffix = schema.testProjectNameSuffix || 'Test'
}
```

I'm not really a fan of that either though... It'd look better if abstracted into a method.
```typescript
// ....
const { suffix, separator } = getPathPartsFromSchema(schema)
// ...

function getPathPartsFromSchema(schema) {
  if (schema.pathScheme === 'nx') {
   return {
    separator: '-'
    suffix: schema.testProjectNameSuffix || 'test'
    }
  } else {
    return {
      separator: '.'
      suffix: schema.testProjectNameSuffix || 'Test'
    }
  }
}
```

**Review by AgentEnder — CHANGES_REQUESTED** [review]
This looks good! Thanks for the contribution 🎉. I left a few comments, once they are fixed we can get it merged.

File: packages/core/src/generators/utils/generate-project.ts
Comment (rcompton78) [reply packages/core/src/generators/utils/generate-project.ts:67]: We should decide what we want the project name to be in workspace.json because if I am not mistaked, the projectName is what is used to add the project to NX.  

So I see a couple of cases that could be NX pathScheme and dotnet.
1- nx g @nx-dotnet/core:library SomeCompany.Foobar --directory=somefolder ...
2- nx g @nx-dotnet/core:library SomeCompany.Foobar ...

In other words, with and without directory

NX Scheme  with directory: somefolder-some-company.foobar,  <directory>-<name>
NX Scheme without directory: some-company.foobar, <name>

And I would assume the following for dotnet scheme
Dotnet Scheme with directory: Somefolder.SomeCompany.Foobar
Dotnet Scheme without directory: SomeCompany.Foobar

What is the reason for adding the directory to the front of the  name?  Just to namespace it?

Feel free to message me directory to discuss.  I am not sure if I was clear here.  I want to get clarity on all cases before I rip this methods apart and try make it cleaner.   That includes making the changes for projectName.

Comment (AgentEnder) [reply packages/core/src/generators/utils/generate-project.ts:67]: Yeah, adding the directory is mainly to namespace it, but also to maintain consistency w/ the other Nx generators. The csproj name matches the generated project name, and w/ the nx first party generators when you specify --directory, it is prepended to the project name.

**Comment by AgentEnder** [comment]
I'm going to merge this now, but going to add a few tests before releasing it.

**Pull request merged**
